lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -451,20 +451,42 @@ static int z_to_nrf_addrinfo_hints(const struct zsock_addrinfo *z_in,
 				   struct nrf_addrinfo *nrf_out)
 {
 	int family;
+	int socktype;
+	int protocol;
 
 	memset(nrf_out, 0, sizeof(struct nrf_addrinfo));
 	nrf_out->ai_flags = z_to_nrf_addrinfo_flags(z_in->ai_flags);
-	nrf_out->ai_socktype = z_to_nrf_socktype(z_in->ai_socktype);
 
 	family = z_to_nrf_family(z_in->ai_family);
 	if (family == -EAFNOSUPPORT) {
 		return -EAFNOSUPPORT;
 	}
 	nrf_out->ai_family = family;
 
-	nrf_out->ai_protocol = z_to_nrf_protocol(z_in->ai_protocol);
-	if (nrf_out->ai_protocol == -EPROTONOSUPPORT) {
-		return -EPROTONOSUPPORT;
+	/* `ai_socktype` can be zero, if not it needs to be a
+	 * supported value
+	 */
+	if (z_in->ai_socktype != 0) {
+		socktype = z_to_nrf_socktype(z_in->ai_socktype);
+		if (socktype == -ESOCKTNOSUPPORT) {
+			return -ESOCKTNOSUPPORT;
+		}
+		nrf_out->ai_socktype = socktype;
+	} else {
+		nrf_out->ai_socktype = 0;
+	}
+
+	/* `ai_protocl` can be zero, if not it needs to be a
+	 * supported value
+	 */
+	if (z_in->ai_protocol != 0) {
+		protocol = z_to_nrf_protocol(z_in->ai_protocol);
+		if (protocol == -EPROTONOSUPPORT) {
+			return -EPROTONOSUPPORT;
+		}
+		nrf_out->ai_protocol = protocol;
+	} else {
+		nrf_out->ai_protocol = 0;
```

Comment:
Why not put this logic in `z_to_nrf_protocol`?

Reply:
@lemrey Because I'm conforming to the preexisting code, also to keep the `z_to_nrf...` functions from becoming bloated.